arrow/arrow.py

```diff
@@ -213,7 +299,9 @@ def utcfromtimestamp(cls, timestamp):
         )
 
     @classmethod
-    def fromdatetime(cls, dt, tzinfo=None):
+    def fromdatetime(
+        cls, dt: Union[dt_datetime, "Arrow"], tzinfo: Optional[TZ_EXPR] = None
```

Comment:
`dt` should be of type `dt_datetime` only. This cannot be used to construct an object from an existing Arrow object.

Reply:
Done! Thanks for catching